src/dstack/_internal/utils/ssh.py

```diff
@@ -1,20 +1,24 @@
 import io
 import os
 import re
+import shutil
 import subprocess
 import sys
 import tempfile
+from dataclasses import dataclass
 from pathlib import Path
-from typing import Dict, Optional
+from typing import Dict, Optional, Tuple, Union
 
 import paramiko
 from filelock import FileLock
 from paramiko.config import SSHConfig
 from paramiko.pkey import PKey, PublicBlob
 from paramiko.ssh_exception import SSHException
 
+from dstack._internal.compat import IS_WINDOWS
+from dstack._internal.core.errors import SSHError
```

Comment:
The current import hierarchy convention is utils<-core<-services, so utils should not import from core (even from errors). In other words, utils should not depend on dstack-specific definitions from other modules. Maybe move new ssh functionality to dstack._internal.services.ssh?